src/common/btree_class.c

```diff
@@ -1444,8 +1444,11 @@ iv_rec_string(struct btr_instance *tins, struct btr_record *rec, bool leaf,
 	void              *k;
 	uint64_t	key;
 
-	if (!leaf) {
-		key_rec = umem_off2ptr(&tins->ti_umm, rec->rec_node[0]);
+	if (!leaf && (tins->ti_root->tr_feats & BTR_FEAT_DIRECT_KEY)) {
```

Comment:
Why need to check "tins->ti_root->tr_feats & BTR_FEAT_DIRECT_KEY"? As my understand, for non-leaf node, nd[0] is always the header to trace current level records, in spite of whether it is direct key or not, right? Seems the API to_rec_string for other classes also have similar issue.

Reply:
For non-direct key, we use the 'rec_hkey' in current btr_record (see following iv_key_get() call). As for other tree classes, I think we don't support direct key for them yet.